addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,1024 @@
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return remoteList;
+    }
+
+    private ByteBuffer sendCommand(String command) {
+
+        if (command != null) {
+            ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
+            try {
+                byteBuffer.put(command.getBytes("ASCII"));
+                onWritable(byteBuffer, socketChannel);
+                Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                return onReadable(socketChannel, ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+            } catch (UnsupportedEncodingException | NumberFormatException | InterruptedException e) {
+                logger.error("An exception occurred while configurting the IRtrans transceiver for '{}': {}",
+                        getThing().getUID(), e.getMessage());
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    private SocketChannel connect(String ipAddress, int portNumber) {
+        SocketChannel socketChannel = null;
+        try {
+            socketChannel = SocketChannel.open();
+        } catch (Exception e2) {
+            logger.error("An exception occurred while connecting to {}:{} : {}",
+                    new Object[] { ipAddress, portNumber, e2.getMessage() });
+        }
+
+        try {
+            socketChannel.socket().setKeepAlive(true);
+            socketChannel.configureBlocking(false);
+        } catch (Exception e) {
+            logger.error("An exception occurred while configuring the connection to '{}:{}' : {}",
+                    new Object[] { ipAddress, portNumber, e.getMessage() });
+        }
+
+        while (selector == null) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                logger.error("An exception occurred while putting the Thread to sleep: {}", e.getMessage(), e);
+            }
+        }
+
+        synchronized (selector) {
+            selector.wakeup();
+            int interestSet = SelectionKey.OP_READ | SelectionKey.OP_WRITE | SelectionKey.OP_CONNECT;
+            try {
+                socketChannelKey = socketChannel.register(selector, interestSet);
+            } catch (ClosedChannelException e1) {
+                logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+            }
+        }
+
+        InetSocketAddress remoteAddress = new InetSocketAddress(ipAddress, portNumber);
+
+        try {
+            logger.trace("Connecting the channel for {} ", remoteAddress);
+            socketChannel.connect(remoteAddress);
+        } catch (Exception e) {
+            logger.error("An exception occurred while connecting connecting to '{}:{}' : {}",
+                    new Object[] { ipAddress, portNumber, e.getMessage() });
+        }
+
+        return socketChannel;
```

Comment:
This method contains 5 try/catches, the idea is that exceptions allow you to separate error handling from the program flow, see also: https://docs.oracle.com/javase/tutorial/essential/exceptions/advantages.html